piccap/js/script.js

```diff
@@ -22,261 +22,150 @@ save()		              - Btn Save
+
+function wait(t) {
+  return new Promise((resolve, reject) => {
+    setTimeout(resolve, t);
+  });
+}
+
+function asyncCall(uri, args) {
+  return new Promise((resolve, reject) => {
+    const service = uri.split('/', 3).join('/');
+    const method = uri.split('/').slice(3).join('/')
+    webOS.service.request(service, {
+      method,
+      parameters: args,
+      onComplete: resolve,
+      onFailure: reject,
     });
+  });
 }
 
-function loadconf(){
-    loadingcounter = 0;
-    console.log("Loading configuration.. Calling getSettings from service.");
-    document.getElementById("servicestatus").innerHTML = "Loading configuration..";
-    webOS.service.request("luna://org.webosbrew.piccap.service/", {
-        method: "getSettings",
-        onFailure: showErrorService,
-        onComplete: makeSuccServiceLoad
-    });
-
-    document.getElementById("servicestatus").innerHTML = "Checking servicestatus..";
-    webOS.service.request("luna://org.webosbrew.piccap.service/", {
-        method: "isStarted",
-        onFailure: showErrorService,
-        onComplete: showSuccServiceisStarted
-    });
-
-    rootStatus();
+async function startup() {
+  await checkRoot();
+  await getSettings();
+  await getStatus();
 }
 
-function save(){
-    document.getElementById("servicestatus").innerHTML = "Saving..";
+startup();
 
-    ip = document.getElementById("ip").value;
-    port = document.getElementById("port").value;
-    // startdelay = document.getElementById("startdelay").value;
-    width = document.getElementById("width").value;
-    height = document.getElementById("height").value;
-    fps = document.getElementById("fps").value;
+async function checkRoot(){
+  let wasRoot = true;
+  let isroot = false;
 
-    if (document.getElementById("radionew").checked){
-        lib = "new";
-    }else if(document.getElementById("radioold").checked){
-        lib = "old";
-    }else{
-        lib = "notChecked";
-    }
-   
-    if(document.getElementById("videocapture").checked){
-        videocapture = "1";
-    }else{
-        videocapture = "0";
-    }
+  for (let i = 0 ; i < 15 ; i++) {
+    const res = await asyncCall('luna://org.webosbrew.piccap.service/isRoot', {});
+    console.info(res);
 
-    if(document.getElementById("graphiccapture").checked){
-        graphiccapture = "1";
-    }else{
-        graphiccapture = "0";
+    if (res.rootStatus){
+      if(!wasRoot){
+        document.getElementById("permissionstatus").innerHTML = 'Evaluate success! Rooted.';
+        await wait(3000);
+        return;
+      }
+      document.getElementById("permissionstatus").innerHTML = 'Running as root';
+      break;
     }
 
-    if(document.getElementById("autostart").checked){
-        autostart = "1";
-    }else{
-        autostart = "0";
+    if(!isroot){
+      wasRoot = false;
+      document.getElementById("permissionstatus").innerHTML = 'Trying to evaluate service..';
```

Comment:
It's called "service elevation" - "bring up" (its permissions), not "evaluate" (judge, rate...) :D

Reply:
Oh yeah, sorry I'm not a native speaker.. :D Corrected it.